test_conformance/extensions/cl_khr_command_buffer/command_buffer_get_command_buffer_info.cpp

```diff
@@ -0,0 +1,340 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "basic_command_buffer.h"
+#include "procs.h"
+
+#include <vector>
+
+//--------------------------------------------------------------------------
+enum class CombufInfoTestMode
+{
+    CITM_QUEUES = 0,
+    CITM_REF_COUNT,
+    CITM_STATE,
+    CITM_PROP_ARRAY,
+};
+
+namespace {
+
+////////////////////////////////////////////////////////////////////////////////
+// clGetCommandBufferInfoKHR tests for cl_khr_command_buffer which handles below
+// cases:
+// -test case for CL_COMMAND_BUFFER_NUM_QUEUES_KHR &
+//  CL_COMMAND_BUFFER_QUEUES_KHR queries
+// -test case for CL_COMMAND_BUFFER_REFERENCE_COUNT_KHR query
+// -test case for CL_COMMAND_BUFFER_STATE_KHR query
+// -test case for CL_COMMAND_BUFFER_PROPERTIES_ARRAY_KHR query
+
+template <CombufInfoTestMode test_mode>
+struct CommandBufferGetCommandBufferInfo : public BasicCommandBufferTest
+{
+    CommandBufferGetCommandBufferInfo(cl_device_id device, cl_context context,
+                                      cl_command_queue queue)
+        : BasicCommandBufferTest(device, context, queue)
+    {}
+
+    //--------------------------------------------------------------------------
+    cl_int Run() override
+    {
+        cl_int error = CL_SUCCESS;
+
+        switch (test_mode)
+        {
+            case CombufInfoTestMode::CITM_QUEUES:
+                error = RunQueuesInfoTest();
+                test_error(error, "RunQueuesInfoTest failed");
+                break;
+            case CombufInfoTestMode::CITM_REF_COUNT:
+                error = RunRefCountInfoTest();
+                test_error(error, "RunRefCountInfoTest failed");
+                break;
+            case CombufInfoTestMode::CITM_STATE:
+                error = RunStateInfoTest();
+                test_error(error, "RunStateInfoTest failed");
+                break;
+            case CombufInfoTestMode::CITM_PROP_ARRAY:
+                error = RunPropArrayInfoTest();
+                test_error(error, "RunPropArrayInfoTest failed");
+                break;
+        }
+
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+    cl_int RecordCommandBuffer()
+    {
+        cl_int error = CL_SUCCESS;
+
+        error = clCommandNDRangeKernelKHR(
+            command_buffer, nullptr, nullptr, kernel, 1, nullptr, &num_elements,
+            nullptr, 0, nullptr, nullptr, nullptr);
+        test_error(error, "clCommandNDRangeKernelKHR failed");
+
+        error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+        return CL_SUCCESS;
+    }
+
+    //--------------------------------------------------------------------------
+#define test_expected_info(cond)                                               \
```

Comment:
Any reason we can't use the existing `test_assert_error` instead of defining a new `test_expected_info`?

Reply:
Done